### DISCUSSION TITLE
Can this app receive "Next filter" broadcasts/intents? How?

### DISCUSSION THREAD

**Discussion by ChaoticNeutralCzech** [LibreShift/red-moon#345] [Q&A] [answered]
**Edit: After reinstalling the app and restarting my device, the problem has been solved so I don't really need the answer anymore.**

----

On my device in Root Mode, Red Moon only starts when another filter is selected, for example when "Next filter" is pressed. I don't care about fixing the issue (it presumably happens only on my rare x86 device), I just want a workaround with [Automation](https://f-droid.org/en/packages/com.jens.automation2/) or another app that automatically "presses Next Filter" when Red Moon is turned on (I already figured out how to receive the RED_MOON_TOGGLED broadcast). I just need somebody who knows about Android intents/broadcasts to tell me how Red Moon can be sent the "Next Filter" action from another app.

AndroidManifest.xml [specifies](https://github.com/LibreShift/red-moon/blob/47a10bbf959db172a44f583c2c53c477cf7603d0/app/src/main/AndroidManifest.xml#L140) `<receiver android:name=".receiver.NextProfileCommandReceiver" />` (not specifying if exported or not) that is further defined in [receiver/NextProfileCommandReceiver.kt](https://github.com/LibreShift/red-moon/blob/47a10bbf959db172a44f583c2c53c477cf7603d0/app/src/main/java/com/jmstudios/redmoon/receiver/NextProfileCommandReceiver.kt). Can someone please tell me what I can type into one of these action setup screen so that the Next Filter action is received by Red Moon? Or do I need to edit and recompile Red Moon to make the receiver exported? At that point I can just try to edit the code to call the "Next filter" action internally whenever the settings tile is pressed.

![intent](https://github.com/user-attachments/assets/6e9f4b64-ef41-4448-a926-992c65292c9b) ![broadcast](https://github.com/user-attachments/assets/4a550d15-a058-4076-9374-37155c43a01d)

**Comment by smichel17** [comment, marked as answer]
I don't really remember, but from a search, I think the answer is you can _toggle_ the filter externally but not change the filter: https://github.com/LibreShift/red-moon/issues/202#issuecomment-748424642

I did want to add more api surface at some point, but nobody ever reached out with a specific use case.

I'm glad you were able to solve this issue in a different way on your own :)
